build_tools/math/mlir_to_filecheck.py

```diff
@@ -0,0 +1,84 @@
+"""Copyright 2024 The StableHLO Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Construct FileCheck comments from a MLIR source.
```

Comment:
@jpienaar is there any existing OSS tooling that does this?

Reply:
Found it -- https://github.com/llvm/llvm-project/blob/main/mlir/utils/generate-test-checks.py

Let's use this tool instead since its standardized.

Note, you may need to mess with the flags to get it working. I was trying with `stablehlo/testdata/add_any_dtypes_lhs_bfloat16_2__rhs_bfloat16_2.mlir` and needed to specify the following options (default soruce_delim is `func @`, that file has multiple functions but only one public, and I'm not sure why starts_from_scope was needed but it was)

```
stablehlo-opt /tmp/t.mlir | generate-test-checks.py --source /tmp/t.mlir --source_delim_regex="func public" --starts_from_scope=0
```

Reply:
Great! Fixed.

The following worked for me:
```sh
build/bin/stablehlo-opt --chlo-legalize-to-stablehlo --split-input-file --verify-diagnostics \
   stablehlo/tests/chlo/chlo_legalize_to_stablehlo.mlir | python llvm-project/mlir/utils/generate-test-checks.py | less
```